File: backend/models/location.go
package models

import (
  "encoding/json"
  "time"
  "github.com/gofrs/uuid"
)

type Location struct {
  ID        uuid.UUID `json:"id" db:"id"`
  CreatedAt time.Time `json:"created_at" db:"created_at"`
  UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
  Address   string    `json:"address" db:"address"`
}

// String is not required by pop and may be deleted
func (b Location) String() string {
  jb, _ := json.Marshal(b)
  return string(jb)
}

// Locations is not required by pop and may be deleted
type Locations []Location

// String is not required by pop and may be deleted
func (b Locations) String() string {
  jb, _ := json.Marshal(b)
  return string(jb)
}

File: backend/src/models/location.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Location = sequelize.define(
    "Location",
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        allowNull: false,
        primaryKey: true
      },
      address: {
        type: DataTypes.STRING,
        unique: true,
        allowNull: false
      },
      approved: {
        type: DataTypes.BOOLEAN,
        defaultValue: false,
        allowNull: false
      },
      createdAt: {
        allowNull: false,
        type: DataTypes.DATE
      },
      updatedAt: {
        allowNull: false,
        type: DataTypes.DATE
      },
      imageURL: {
        type: DataTypes.STRING,
      }
    },
    {}
  );
  Location.associate = function(models) {
    models.Location.hasMany(models.Suggestion, { onDelete: "CASCADE" });
  };
  return Location;
};
